lib/bridge/IrcHandler.js

```diff
@@ -176,6 +185,86 @@ IrcHandler.prototype.onPrivateMessage = Promise.coroutine(function*(req, server,
     yield this.ircBridge.sendMatrixAction(pmRoom, virtualMatrixUser, mxAction, req);
 });
 
+IrcHandler.prototype._serviceTopicQueue = Promise.coroutine(function*(item) {
+    let promises = item.entries.map((entry) => {
+        if (entry.matrix.topic === item.topic) {
+            item.req.log.info(
+                "Topic of %s already set to '%s'",
+                entry.matrix.getId(),
+                item.topic
+            );
+            return Promise.resolve();
+        }
+        console.log('setRoomTopic');
+        return this.ircBridge.getAppServiceBridge().getIntent().setRoomTopic(
+            entry.matrix.getId(),
+            item.topic
+        ).then(
+            () => {
+                console.log('upsertRoomStoreentry');
+                entry.matrix.topic = item.topic;
+                return this.ircBridge.getStore().upsertRoomStoreEntry(entry);
+            },
+            (err) => {
+                item.req.log.error(`Error storing room ${entry.matrix.getId()} (${err.message})`);
+            }
+        );
+    });
+    try {
+        yield Promise.all(promises);
+        item.req.log.info(
+            "Set topics: ",
+            item.entries.map(
+                (entry) => `id = ${entry.matrix.getId()}, topic = '${entry.matrix.topic}'`
+            ).join('\n')
```

Comment:
The topic will always be the same, so no point logging it N times where N = the number of rooms. Also, the topic may be large, so I would cheat and substring it to death via `.substring(0, 20)` (first 20 chars) which will Do The Right Thing if the topic is < 20 chars too. So in total, something like this please:
```js
item.req.log.info(
    `Topic '${item.topic.substring(0, 20)}...' set in rooms: `, 
    item.entries.map((entry) => entry.matrix.getId()).join(",")
);
```